OnProfileUpdate/handler.ts

```diff
@@ -0,0 +1,188 @@
+  );
+
+const insertProfileEmail = (profileEmail: ProfileEmail) => ({
+  dataTableProfileEmailsRepository
+}: IDependencies): TE.TaskEither<Error, void> =>
+  pipe(
+    TE.tryCatch(
+      () => dataTableProfileEmailsRepository.insert(profileEmail),
+      error => error
+    ),
+    TE.orElse(insertError =>
+      TE.tryCatch(
+        () =>
+          // check if the insert operation failed because the record was already there (for example in case of retry of the entire batch)
+          dataTableProfileEmailsRepository.get(profileEmail).then(() => {}),
+        () =>
+          insertError instanceof Error
+            ? insertError
+            : new Error("error inserting ProfileEmail into table storage")
+      )
+    )
+  );
+
+/*
+If the current email is validated but the previous email was not validated => it inserts the new email into profileEmails
+If the current email is not validated but the previous email was validated => it deletes the previous email from profileEmails
+*/
+const handlePositiveVersion = ({
+  email,
+  fiscalCode,
+  isEmailValidated,
+  version,
+  _self
+}: ProfileDocument): RTE.ReaderTaskEither<
+  IDependencies,
+  Error | CosmosErrors,
+  void
+> =>
+  pipe(
+    getPreviousProfile(fiscalCode, version),
+    RTE.chainW(
+      flow(
+        O.foldW(
+          () =>
+            pipe(
+              RTE.asks(({ logger }: IDependencies) =>
+                logger.error(
+                  `no previous profile found for profile with _self ${_self}`
+                )
+              ),
+              RTE.map(() => void 0)
+            ),
+          previousProfile =>
+            isEmailValidated
+              ? previousProfile.isEmailValidated
+                ? RTE.right(void 0)
+                : insertProfileEmail({ email, fiscalCode })
+              : previousProfile.isEmailValidated
+              ? deleteProfileEmail({
+                  email: previousProfile.email,
+                  fiscalCode
+                })
+              : RTE.right(void 0)
+        )
+      )
+    )
+  );
+
+const handleProfile = (
+  profile: ProfileDocument
+): RTE.ReaderTaskEither<IDependencies, Error | CosmosErrors, void> =>
+  profile.version === 0
+    ? profile.isEmailValidated
+      ? insertProfileEmail({
+          email: profile.email,
+          fiscalCode: profile.fiscalCode
+        })
+      : RTE.right<IDependencies, Error, void>(void 0)
+    : handlePositiveVersion(profile);
+
+export const handler = (documents: ReadonlyArray<unknown>) => (
+  dependencies: IDependencies
+): T.Task<ReadonlyArray<E.Either<Error | CosmosErrors, void>>> =>
+  pipe(
+    documents,
+    A.map(document =>
+      pipe(
+        document,
+        ProfileDocument.decode,
+        E.foldW(
+          errors => {
+            dependencies.logger.error(
+              `error decoding profile with errors ${PathReporter.report(
+                E.left(errors)
+              )}`
+            );
```

Comment:
The document id that fail the decode is importart.
```suggestion
            dependencies.logger.error(
              `error decoding profile with errors ${PathReporter.report(
                E.left(errors)
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
              )} [${(document && (document as any).id) || "UNKNOWN ID"}]`
            );
```

Reply:
Actually the `id` includes the fiscal code